num_enum_derive/src/lib.rs

```diff
@@ -119,77 +119,75 @@ struct VariantInfo {
     ident: Ident,
     attr_spans: AttributeSpans,
     is_default: bool,
+    is_complex: bool,
     canonical_value: Expr,
     alternative_values: Vec<Expr>,
 }
 
+impl VariantInfo {
+    fn all_values(&self) -> impl Iterator<Item = &Expr> {
+        ::core::iter::once(&self.canonical_value).chain(self.alternative_values.iter())
+    }
+}
+
 struct EnumInfo {
     name: Ident,
     repr: Ident,
     variants: Vec<VariantInfo>,
 }
 
-struct CanonicalAndAlternatives<T> {
-    canonical: Vec<T>,
-    alternatives: Vec<Vec<T>>,
-}
-
 impl EnumInfo {
-    fn has_default(&self) -> bool {
+    fn has_default_variant(&self) -> bool {
         self.default().is_some()
     }
 
-    fn has_alternatives(&self) -> bool {
-        self.variants.iter().any(|info| !info.alternative_values.is_empty())
+    fn has_complex_variant(&self) -> bool {
+        self.variants.iter().any(|info| info.is_complex)
```

Comment:
I feel like the whole `is_complex` extra variable isn't worth it, what about inlining the check here?

```suggestion
        self.variants.iter().any(|info| ! info.alternative_values.is_empty())
```

Reply:
Good point. Fixed.